tests/rustdoc/synthetic_auto/no-redundancy.rs

```diff
@@ -10,7 +10,7 @@ where
 
 // @has no_redundancy/struct.Outer.html
 // @has - '//*[@id="synthetic-implementations-list"]//*[@class="impl"]//h3[@class="code-header"]' \
-// "impl<T> Send for Outer<T>where T: Send + Copy"
+// "impl<T> Send for Outer<T>where T: Copy + Send"
```

Comment:
Why does this fix a test?

Reply:
Me being quite familiar with `src/librustdoc/clean/, I presume that this is rustdoc's fault.
rustdoc still uses a lot of `FxHash{Set,Map}`s in its code base (20 of them in total in `clean/auto_trait.rs` which is responsible for generating this `Send` impl).

I'm not sure why exactly it's manifests like that but `FxHash{Set,Map}` (while *stable* under re-runs & recompilations of rustdoc) is “*unstable* under `rmeta` changes”. Since this PR has modified `core`, `core`'s `rmeta` file has changed and rustdoc being rustdoc now generates different output. Mind you, this will stay like that, **it's not a flaky test** (at least if my theory checks out).

To give you another example, in PR #116388 I was able to get rid of several cases of this “instability” in rustdoc. Check out the *Examples* `<details/>` in the PR description.

---

@compiler-errors, since this is a bug in rustdoc and we can't do anything short-term here, this PR shouldn't be blocked on it. I officially approve the rustdoc test changes, so to say ^^.

Reply:
I've just opened #119597 for that.